### PR TITLE
Add rate limited logging.

### PR DESCRIPTION
**Pull request by vjkoskela**
Addresses #12, will likely have a few minor merge conflicts with #13 .

### CODE REVIEW

**Comment by vjkoskela** [comment]
@BrandonArp rebased and ready for review.

File: src/main/java/com/arpnetworking/metrics/impl/ApacheHttpSink.java [hunk 1]
```diff
@@ -119,8 +127,14 @@ private ApacheHttpSink(final Builder builder) {
 
     private final class HttpDispatch implements Runnable {
 
-        /* package private */ HttpDispatch(final PoolingHttpClientConnectionManager connectionManager) {
-            _connectionManager = connectionManager;
+        /* package private */ HttpDispatch(
+                final PoolingHttpClientConnectionManager connectionManager,
+                final Duration dispatchErrorLoggingInterval) {
+                _connectionManager = connectionManager;
+             _dispatchErrorLogger = new RateLimitedLogger(
```
Comment (BrandonArp) [inline src/main/java/com/arpnetworking/metrics/impl/ApacheHttpSink.java]: I don't think this is quite right.  You're going to create one of these on each thread.  Your actual rate will be multiplied by the number of threads running.

Comment (vjkoskela) [inline src/main/java/com/arpnetworking/metrics/impl/ApacheHttpSink.java]: That's no longer the case after we merged #13 which creates a single HttpDispatch instance for the entire worker pool. So while independent ApacheHttpSink instances will have independent rate limits within each there should be a singular rate limiter for the thread pool. Am I missing something here?

Comment (BrandonArp) [inline src/main/java/com/arpnetworking/metrics/impl/ApacheHttpSink.java]: Yep, you're right.  Sorry.

**Pull request merged**